ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryController.java

```diff
@@ -411,6 +430,57 @@ public synchronized void reset() {
         db = null;
     }
 
+    /**
+     * Checks if the datasources table in drawable DB is stale.
+     * 
+     * @return true if datasources table is stale
+     */
+    boolean isDataSourcesTableStale() {
+        
+        // no current case open to check
+        if ((null == db) || (null == sleuthKitCase)) {
+            return false;
+        }
+        
+        try {
+            Set<Long> knownDataSourceIds= db.getKnownDataSourceIds();
+            List<DataSource> dataSources = sleuthKitCase.getDataSources();
+            Set<Long> caseDataSourceIds = new HashSet<>();
+            dataSources.forEach((dataSource) -> {
+                caseDataSourceIds.add(dataSource.getId());
```

Comment:
I would do this as 

Set<Long> caseDataSourceIds = dataSources.map(DataSource::getId).collect(Collectors.toSet());

But that is just me.